.github/workflows/nightly.yaml

```diff
@@ -7,7 +7,7 @@ jobs:
   nightly:
     strategy:
       matrix:
-        go-version: [1.19.x]
+        go-version: [1.17.x, 1.23.x]
```

Comment:
I'm ignorant of how all this github CI stuff works, but from what I'm reading at https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow, this means that it will build artifacts *both* with 1.17 and 1.23? We might want that for tests (`test.yaml`), but probably not for releases and nightlies? For them, I guess we should specify just the version we actually want releases to be built with, e.g. 1.23?

Reply:
> [...] this means that it will build artifacts both with 1.17 and 1.23?

Yes, using multiple entries will cycle over them.

> We might want that for tests (`test.yaml`), but probably not for releases and nightlies?

Correct, for the tests this idea wasn't bad, but for the nighly and release it is counterproductive.

> For them, I guess we should specify just the version we actually want releases to be built with, e.g. 1.23?

Agree. We should use the latest stable.